orttraining/orttraining/training_api/onnxruntime_training_c_api.cc

```diff
@@ -184,6 +190,17 @@ ORT_API_STATUS_IMPL(OrtTrainingApis::EvalStep, _In_ const OrtTrainingSession* se
   API_IMPL_END
 }
 
+ORT_API_STATUS_IMPL(OrtTrainingApis::SetLearningRate, _Inout_ OrtTrainingSession* sess,
+                    _In_ float learning_rate) {
+  API_IMPL_BEGIN
+
+  auto session = reinterpret_cast<onnxruntime::training::api::TrainingSession*>(sess);
+  ORT_API_RETURN_IF_STATUS_NOT_OK(SetLearningRateImpl(session, learning_rate));
```

Comment:
Sorry, get back here again. :)

I think this is still doing the same thing in term of  setting learning rate as RegisterLRScheduler did when initial_lr is given. 
They are both operating on internal initial_lr. So shall we change the parameter name "learning_rate" to "initial_lr" for alignment.

Reply:
OKay, I know the gap now. The private SetLearningRate in orttraining/orttraining/training_api/include/optimizer.h is internally used by lr_shcudler during Step(), the parameter it takes is the lr after calculation based on initial_lr and some factors. 

Here, when we register LR with an initial_lr, we want to set initial_lr. So we need a public  SetInitialLearningRate(float initial_lr) interface for Optimizer. 

I would suggest for OrtTrainingApis::SetLearningRate, we also set the initial lr, I believe when user want to set lr, they usually means the initial lr.

Reply:
Seems like the initial learning rate only matters for pre-defined lr schedulers. What we can do is allow users to set the initial learning rate when registering the learning rate scheduler using the `RegisterLRScheduler` api.
And if they want to set the learning rate (not the initial one), they can use the `SetLearningRate` api for scenarios where they have custom lr schedulers.

Reply:
Okay, that means, for user defined lr scheduler, they can use SetLearningRate, does it imply RegisterLRScheduler is not needed for this kind of usage?

Reply:
Yes, in case users are using their own LR scheduler, they no longer need to register the learning rate scheduler. They must call `SetLearningRate` based on the custom scheduler they are using.

Reply:
Please kindly help add some pseudo-code for how the code looks like when they wanted to use customized lr scheduler.